src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml

```diff
@@ -0,0 +1,205 @@
+    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="Template">
+      <Setter.Value>
+        <ControlTemplate>
+          <Grid>
+            <Grid.ColumnDefinitions>
+              <ColumnDefinition Width="*" />
+              <ColumnDefinition Width="Auto" />
+            </Grid.ColumnDefinitions>
+
+            <Button x:Name="PART_PrimaryButton"
+                    Grid.Column="0"
+                    MinWidth="{DynamicResource SplitButtonPrimaryButtonSize}"
+                    Foreground="{TemplateBinding Foreground}"
+                    Background="{TemplateBinding Background}"
+                    BorderThickness="{TemplateBinding BorderThickness}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    Content="{TemplateBinding Content}"
+                    ContentTemplate="{TemplateBinding ContentTemplate}"
+                    Command="{TemplateBinding Command}"
+                    CommandParameter="{TemplateBinding CommandParameter}"
+                    FontFamily="{TemplateBinding FontFamily}"
+                    FontSize="{TemplateBinding FontSize}"
+                    FontWeight="{TemplateBinding FontWeight}"
+                    HorizontalAlignment="Stretch"
+                    VerticalAlignment="Stretch"
+                    HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
+                    VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}"
+                    Padding="{TemplateBinding Padding}"
+                    Focusable="False"
+                    KeyboardNavigation.IsTabStop="False" />
+
+            <Button x:Name="PART_SecondaryButton"
+                    Grid.Column="1"
+                    MinWidth="{DynamicResource SplitButtonSecondaryButtonSize}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    Foreground="{TemplateBinding Foreground}"
+                    Background="{TemplateBinding Background}"
+                    BorderThickness="{TemplateBinding BorderThickness}"
+                    HorizontalContentAlignment="Right"
+                    VerticalContentAlignment="Center"
+                    HorizontalAlignment="Stretch"
+                    VerticalAlignment="Stretch"
+                    Focusable="False"
+                    KeyboardNavigation.IsTabStop="False" />
+          </Grid>
+        </ControlTemplate>
+      </Setter.Value>
+    </Setter>
+  </Style>
+
+  <!-- Default overridable styles -->
+  <Style Selector="SplitButton /template/ Button#PART_PrimaryButton">
+    <Setter Property="CornerRadius" Value="{TemplateBinding CornerRadius, Converter={StaticResource LeftCornerRadiusFilterConverter}}" />
+  </Style>
+  <Style Selector="SplitButton /template/ Button#PART_SecondaryButton">
+    <Setter Property="CornerRadius" Value="{TemplateBinding CornerRadius, Converter={StaticResource RightCornerRadiusFilterConverter}}" />
```

Comment:
> Apparently trick with CornerRadius can be done here easily. It's less hacky then clipping.

Yes... I know... it was in my latest changes you overwrote without reviewing.

Reply:
I did reviewed it while redoing some changes on top of latest commit, though my comment was written before that.

Reply:
Main difference is between setting this CornerRadius inside of the template, or outside with style setter. While first one is fine for me, I went with second allowing better customability. Unfortunately you can't override binding outside of the template with a style.

Reply:
Simple scenario, when developer wants to have custom corner radiuses set on these buttons directly, make both of them rounded. Can't be done with single SplitButton.CornerRadius, but can be done with direct selectors.

Reply:
Seems overly specific and I would let them re-template for that. It also violates the Fluent design guidelines. I'll let you have your way here though. It's just not something I would do and I think it's an unnecessary complexity and extremely small performance hit.

Reply:
I checked UWP Fluent v2. The border does seem to go around the entire control but not the middle separator. So your changes don't support this.

![image](https://user-images.githubusercontent.com/17993847/151249460-10ff1d2c-15a7-4e88-a0eb-f1c8abe64e35.png)

![image](https://user-images.githubusercontent.com/17993847/151249691-e5473b44-ed51-42ef-8226-a0b54ea34d78.png)

![image](https://user-images.githubusercontent.com/17993847/151249596-a1aaefed-aef6-4b53-b1c5-bb82abff65ea.png)